src/model/Entity/Entity.php

```diff
@@ -0,0 +1,26 @@
+<?php
+namespace base\model\entity;
+
+class Entity
+{
+	public $attributes;
+	protected $name;
+	protected $repositoryName = null;
+	
+	public function getRepositoryName()
+	{
+		return $this->repositoryName;
+	}
+	public function setRepositoryName($repositoryName)
```

Comment:
Does $repository have an interface to avoid unknown $repository structure, and maybe play with object instead of string